### ISSUE TITLE
Walberla: LBFluidWalberla lattice argument

### ISSUE THREAD

**Issue by reinaual** [espressomd/espresso#4421]
This is a minor bug, the `LBFLuidWalberla` constructor allows the `LatticeWalberla` to be provided with the lattice argument but also always wants the `agrid` argument which would not be necessary. This allows for user-errors like this
```python
import espressomd
import espressomd.lb

system = espressomd.System(box_l=[10, 10, 10])
system.time_step = 1.0

lattice = espressomd.lb.LatticeWalberla(n_ghost_layers=1, agrid=1.0)

lbf = espressomd.lb.LBFluidWalberla(lattice=lattice, viscosity=1.0, density=1.0, tau=1.0, agrid=2.0)

system.actors.add(lbf)
```

which are then caught in the `lb_walberla_instance.cpp:lb_sanity_checks` with the error
```
ERROR: 
MPI rank 0: left ESPResSo: [0, 0, 0], left waLBerla: [0, 0, 0]
MPI rank 0: right ESPResSo: [10, 10, 10], right waLBerla: [20, 20, 20]
ERROR: during waLBerla activation: waLBerla and ESPResSo disagree about domain decomposition.
Traceback (most recent call last):
  File "../script/lb_issue.py", line 11, in <module>
    system.actors.add(lbf)
  File "actors.pyx", line 218, in espressomd.actors.Actors.add
  File "/ssd/areinauer/espresso/rudolf/build/src/python/espressomd/lb.py", line 108, in _activate
    self._activate_method()
  File "/ssd/areinauer/espresso/rudolf/build/src/python/espressomd/lb.py", line 115, in _activate_method
    self.call_method('activate')
  File "script_interface.pyx", line 153, in espressomd.script_interface.PScriptInterface.call_method
  File "utils.pyx", line 236, in espressomd.utils.handle_errors
  File "utils.pyx", line 255, in espressomd.utils.handle_errors
Exception: : ERROR: 
MPI rank 0: left ESPResSo: [0, 0, 0], left waLBerla: [0, 0, 0]
MPI rank 0: right ESPResSo: [10, 10, 10], right waLBerla: [20, 20, 20]
```

This obviously is not an issue if the `agrid` value is set correctly.

**Comment by jngrad** [comment]
This is a limitation of the actor class. The list of valid keys must be a tuple or set, which prevents us from specifying that the constructor can either accept a `LatticeWaberla` or an `agrid`/`tau` pair. One solution would be to make it mandatory to initialize all LB actors with a Lattice object instead of an `agrid`/`tau` pair. This would also better reflect what the core actually does.

**Comment by jngrad** [comment]
This should be fixed by 5bdb614fbe595b67. Users can now provide either `lattice` or `agrid`, but not both.
